### PR TITLE
feat: Time how long it takes to load document metadata

### PR DESCRIPTION
**Pull request by arkadyan**
Asana ticket: [Log time it takes to load document metadata](https://app.asana.com/0/1199957631529739/1200373916485244)

### CODE REVIEW

File: lib/catalog.ex [hunk 1]
```diff
@@ -54,6 +56,9 @@ defmodule Catalog do
   def handle_continue({:load_documents, opts}, %__MODULE__{ets: ets} = state) do
     documents_fn = Keyword.get(opts, :documents_fn, &all_documents/0)
 
+    start_time = Time.utc_now()
```
Comment (paulswartz) [inline lib/catalog.ex:59]: You probably want `System.monotonic_time(:millisecond)` instead. This avoids any possible issue where the time on the computer changes while this is measurement is happening.

The other option is to use `:timer.tc` which takes a function and return `{time, return_value}`.

Comment (arkadyan) [reply lib/catalog.ex]: Oh great, thanks. Refactoring out a function to contain the loop and using `:timer.tc` seemed nice and clean so I went that route.

File: lib/catalog.ex [hunk 2]
```diff
@@ -55,26 +55,17 @@ defmodule Catalog do
...
+    {time_in_µs, _} = :timer.tc(&populate_ets_table/2, [ets, documents])
 
     # Let the health server know that the data has finished loading
     Catalog.Health.loaded()
 
     Logger.info(fn ->
-      "Finished loading document metadata, time_in_ms=#{
-        Time.diff(Time.utc_now(), start_time, :millisecond)
-      }"
+      "Finished loading document metadata, time_in_ms=#{time_in_µs / 1_000}"
```
Comment (paulswartz) [inline lib/catalog.ex:68]: FWIW, this is now slightly different: the previous behavior included the time of `document_fn` in the reported time. Not sure if that's what you want or not.

File: lib/catalog.ex [hunk 3]
```diff
@@ -127,6 +118,18 @@ defmodule Catalog do
...
+  @spec populate_ets_table(:ets.tid(), [t()]) :: [t()]
+  defp populate_ets_table(ets, documents) do
+    for document <- documents do
+      _ =
+        :ets.insert(
+          ets,
+          {Document.key(document), document.last_name, document.first_name,
+           document.date_of_birth, document}
+        )
+    end
```
Comment (paulswartz) [inline lib/catalog.ex:130]: `:ets.insert` can also take a list, so you could do a map and then insert the documents all at once:
```elixir
records = for document <- documents do
 {Document.key(document), document.last_name, document.first_name, document.date_of_birth, document}
end
:ets.insert(ets, records)
```

**Review by paulswartz — APPROVED** [review]
small notes, otherwise 🍰

File: lib/catalog.ex (hunk 2 above)
Comment (arkadyan) [reply lib/catalog.ex:67]: Oh good catch! I definitely do want to include that.

**Pull request merged**